### PR TITLE
[CBRD-20217] Extend update undo to check if the undo record should be vacuumed

### PR DESCRIPTION
**Pull request by andrei14vl**
http://jira.cubrid.org/browse/CBRD-20217

The bug was caused by this events:
1) insert record A and commit
2.0) update record A to record B
3) vacuum: it expects record A but finds record B; does not vacuum the record
2.1) rollback update
4) checkdb

The 2.1) rollback will put back record A, with its INSERT ID unvacuumed, which is not acceptable for checkdb. The record will appear as skipped by vacuum.

The fix will handle this case by checking the record header at undo and remove INSERT ID and prev version if necessary.

### CODE REVIEW

**Comment by ribram** [comment]
@andrei14vl 

#### vacuum_check_record_at_undoredo
1 should be used only at undo and should be renamed as vacuum_rv_check_at_undo (or clean on undo, you pick).
2 mvcc_satisfies_vacuum last argument should be vacuum_Global_oldest_active_mvccid.

#### heap_rv_redo_insert
1 this is not the right place to call vacuum_check_record_at_undoredo . it should be called in heap_rv_undo_delete.

#### heap_rv_undoredo_update 
1. if update physical fails, just assert_release (false) and return ER_FAILED;
2. limit the check for undo only... wrap the function with two other function, heap_rv_redo_update and heap_rv_undo_update. check vacuum only in heap_rv_undo_update.
3. if vacuum check fails, assert_release (false) and return ER_FAILED;

**Comment by ribram** [comment]
@andrei14vl 

also heap_rv_undoredo_update_and_update_chain can be renamed as heap_rv_redo_update_and_update_chain

File: src/storage/heap_file.c [hunk 1]
```diff
@@ -16305,7 +16305,73 @@ heap_rv_redo_mark_reusable_slot (THREAD_ENTRY * thread_p, LOG_RCV * rcv)
...
+  error_code = heap_rv_undoredo_update (thread_p, rcv);
+  if (error_code != NO_ERROR)
+    {
+      ASSERT_ERROR ();
+      return error_code;
+    }
+
+  /* vacuum atomicity */
+  recdes_type = *(INT16 *) (rcv->data);
+  if (recdes_type == REC_HOME || recdes_type == REC_RELOCATION)
```
Comment (ribram) [inline src/storage/heap_file.c]: > if (recdes_type == REC_HOME || recdes_type == REC_RELOCATION)

should be
```c
if (recdes_type == REC_HOME || recdes_type == REC_NEWHOME)
```

**Comment by eseokoh** [comment]
@andrei14vl 

It looks good to me. :)

**Comment by eseokoh** [comment]
@andrei14vl 

ah.. It should. :)

**Pull request merged**